Jellyfin.Api/Filters/ItemAccessFilter.cs

```diff
@@ -0,0 +1,115 @@
+using System;
+using Jellyfin.Api.Extensions;
+using Jellyfin.Api.Helpers;
+using Jellyfin.Extensions;
+using MediaBrowser.Controller.Entities;
+using MediaBrowser.Controller.Library;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Extensions.Logging;
+
+namespace Jellyfin.Api.Filters;
+
+/// <summary>
+/// Action filter to assess whether a user can access an item.
+/// </summary>
+public sealed class ItemAccessFilter : IActionFilter
+{
+    private const string ItemIdArgument = "itemId";
```

Comment:
i am not too stoked for this to be name based. To much "magic" involved if you just read the controller side.